msgraph/models.go

```diff
@@ -693,12 +693,66 @@ type ConditionalAccessSessionControls struct {
 }
 
 type ConditionalAccessUsers struct {
-	IncludeUsers  *[]string `json:"includeUsers,omitempty"`
-	ExcludeUsers  *[]string `json:"excludeUsers,omitempty"`
-	IncludeGroups *[]string `json:"includeGroups,omitempty"`
-	ExcludeGroups *[]string `json:"excludeGroups,omitempty"`
-	IncludeRoles  *[]string `json:"includeRoles,omitempty"`
-	ExcludeRoles  *[]string `json:"excludeRoles,omitempty"`
+	IncludeUsers                 *[]string                               `json:"includeUsers,omitempty"`
+	ExcludeUsers                 *[]string                               `json:"excludeUsers,omitempty"`
+	IncludeGroups                *[]string                               `json:"includeGroups,omitempty"`
+	ExcludeGroups                *[]string                               `json:"excludeGroups,omitempty"`
+	IncludeRoles                 *[]string                               `json:"includeRoles,omitempty"`
+	ExcludeRoles                 *[]string                               `json:"excludeRoles,omitempty"`
+	IncludeGuestsOrExternalUsers *ConditionalAccessGuestsOrExternalUsers `json:"includeGuestsOrExternalUsers"`
+	ExcludeGuestsOrExternalUsers *ConditionalAccessGuestsOrExternalUsers `json:"excludeGuestsOrExternalUsers"`
+}
+
+type ConditionalAccessGuestsOrExternalUsers struct {
+	GuestOrExternalUserTypes *[]ConditionalAccessGuestOrExternalUserType `json:"guestOrExternalUserTypes,omitempty"`
```

Comment:
When we're using a custom marshaler and the field type is different, we should tag it to be ignored by default, then it won't accidentally get marshaled to the wrong type if someone embeds it.

```suggestion
	GuestOrExternalUserTypes *[]ConditionalAccessGuestOrExternalUserType `json:"-"` // see ConditionalAccessGuestsOrExternalUsers.MarshalJSON / ConditionalAccessGuestsOrExternalUsers.UnmarshalJSON
```